diff --git a/dist/index.mjs b/dist/index.mjs
index 6855ae0ad..093333e09 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -5 +5 @@ import fs from 'node:fs';
-import StreamArray from 'stream-json/streamers/StreamArray.js';
+import StreamArray from 'stream-json/streamers/stream-array.js';
@@ -990 +990 @@ function jsonlReadFile(filePath) {
-  }).pipe(StreamArray.withParser()).pipe(new MapValues(filePath));
+  }).pipe(StreamArray.withParserAsStream()).pipe(new MapValues(filePath));
diff --git a/dist/index.umd.js b/dist/index.umd.js
index 26b4878e2..4217264b1 100644
--- a/dist/index.umd.js
+++ b/dist/index.umd.js
@@ -2,2 +2,2 @@
-  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('node:stream'), require('@flatten-js/interval-tree'), require('trace-event-lib'), require('node:fs'), require('stream-json/streamers/StreamArray.js'), require('multi-sort-stream')) :
-  typeof define === 'function' && define.amd ? define(['exports', 'node:stream', '@flatten-js/interval-tree', 'trace-event-lib', 'node:fs', 'stream-json/streamers/StreamArray.js', 'multi-sort-stream'], factory) :
+  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('node:stream'), require('@flatten-js/interval-tree'), require('trace-event-lib'), require('node:fs'), require('stream-json/streamers/stream-array.js'), require('multi-sort-stream')) :
+  typeof define === 'function' && define.amd ? define(['exports', 'node:stream', '@flatten-js/interval-tree', 'trace-event-lib', 'node:fs', 'stream-json/streamers/stream-array.js', 'multi-sort-stream'], factory) :
@@ -1014 +1014 @@
-    }).pipe(StreamArray__default["default"].withParser()).pipe(new MapValues(filePath));
+    }).pipe(StreamArray__default["default"].withParserAsStream()).pipe(new MapValues(filePath));
diff --git a/src/streams/jsonl/jsonlReadFile.ts b/src/streams/jsonl/jsonlReadFile.ts
index 480c34983..016599497 100644
--- a/src/streams/jsonl/jsonlReadFile.ts
+++ b/src/streams/jsonl/jsonlReadFile.ts
@@ -5 +5 @@ import { Transform } from 'node:stream';
-import StreamArray from 'stream-json/streamers/StreamArray.js';
+import StreamArray from 'stream-json/streamers/stream-array.js';
@@ -10 +10 @@ export function jsonlReadFile(filePath: string): Readable {
-    .pipe(StreamArray.withParser())
+    .pipe(StreamArray.withParserAsStream())
