diff --git a/src/logger/utils/streams/DetoxJSONLParser.js b/src/logger/utils/streams/DetoxJSONLParser.js
index d94743237..e2fc55043 100644
--- a/src/logger/utils/streams/DetoxJSONLParser.js
+++ b/src/logger/utils/streams/DetoxJSONLParser.js
@@ -1 +1 @@
-const JsonlParser = require('stream-json/jsonl/Parser');
+const JsonlParser = require('stream-json/jsonl/parser.js').default;
@@ -13 +13 @@ class DetoxJSONLParser {
-    const writable = JsonlParser.make({ checkErrors: true })
+    const writable = JsonlParser.asStream({ checkErrors: true })
